src/Processors/Formats/Impl/CSVRowInputFormat.cpp

```diff
@@ -36,6 +36,86 @@ CSVRowInputFormat::CSVRowInputFormat(
             ErrorCodes::BAD_ARGUMENTS);
 }
 
+void CSVRowInputFormat::readPrefix()
+{
```

Comment:
@Avogar please take a look at this change in `CSVRowInputFormat`. Looks like it might intersect with some of your tasks, may be you have something to say about it.

Reply:
This code is almost a complete copy paste of the code from `readPrefix` in `RowInputFormatWithNamesAndTypes` - that's not good. And I see that you initialize csv format settings in StorageHive and it uses input_field_names (we need this setting for columns mapping?). It's not good to change CSV format to fit it for Hive, we should create a new special format (maybe `HiveText`), inherited from CSV (with_names=true and with_types=false, method readNames will read names from setting input_field_names, method readTypes should throw an exception), it will also have it's own format settings.

Reply:
Your design sounds good, I will finish it soon.

Reply:
done. see format `HiveText`